readthedocs/search/api.py

```diff
@@ -78,13 +119,7 @@ def get_queryset(self):
         query = self.request.query_params.get('q', '')
         kwargs = {'filter_by_user': False, 'filters': {}}
         kwargs['filters']['project'] = [p.slug for p in self.get_all_projects()]
-        kwargs['filters']['version'] = self.request.query_params.get('version')
-        if not kwargs['filters']['project']:
-            log.info("Unable to find a project to search")
-            return HTMLFile.objects.none()
-        if not kwargs['filters']['version']:
-            log.info("Unable to find a version to search")
-            return HTMLFile.objects.none()
```

Comment:
Why did we remove this validation? It is important that we keep it to avoid another security issue like the last time we messed this up.

Reply:
This block happens when project or version are None, if project or version are None, a 404 is raised in the code above. So this block isn't needed.

Reply:
What happens if `get_all_projects` returns an empty list?

Reply:
Code doesn't just get added to our repos for no reason. You really need to be careful when deleting or changing things, and understand _why_ they were there. This would likely have caused another major security issue with search if it was shipped.

Reply:
I can't think of a case where it returns empty, but I'm adding a test mocking it to return empty

Reply:
Where a project doesn't have any public versions and the user doesn't have access to a private version. It happened in the past, and maybe the code is more defensive now, but I'd really like to keep this additional check just to be sure.

Reply:
Check is back with a test!